### PR TITLE
Add `ExternalAccess.RazorCompiler.dll` reference to C# language server

### PR DESCRIPTION
**Pull request by allisonchou**
Razor relies on Roslyn to ship this dll with the language server since they use it for source generators.

### CODE REVIEW

**Comment by genlu** [comment]
Shouldn't this be shipped as part of Razor then?

**Comment by allisonchou** [comment]
> Shouldn't this be shipped as part of Razor then?

Roslyn's `dotnet.exe` process is the one attempting to load the dll in VS Code. I confirmed with David that it'd be ok to include as part of the C# language server

**Comment by 333fred** [comment]
> Shouldn't this be shipped as part of Razor then?

No. We want the version of the external access to match the shipped version of roslyn, not razor, as underlying API changes happen but we keep the EA layer stable. I'm sure @sharwell would also be happy to elaborate on the reasoning.

**Review by 333fred — APPROVED** [review]

**Review by dibarbet — APPROVED** [review]

**Comment by sharwell** [comment]
> Shouldn't this be shipped as part of Razor then?

Absolutely not. This package is version-locked to a _single exact_ version of the compiler (no deviation is allowed in either direction), and therefore must ship with the compiler.

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj [hunk 1]
```diff
@@ -58,6 +58,9 @@
 
     <!-- Dlls we don't directly reference but need to include to build the MEF composition -->
     <ProjectReference Include="..\..\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.Features.csproj" />
+
+    <!-- Not directly referenced but needed for Razor source generators -->
+    <ProjectReference Include="..\..\..\Tools\ExternalAccess\RazorCompiler\Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj" />
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj:63]: @allisonchou This needs `ReferenceOutputAssembly="false"` added

Comment (allisonchou) [reply src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj:63]: Got it, will add it in a follow up PR

**Pull request merged**